### ISSUE TITLE
"Unable to get page count.\n%s" % err.decode("utf8", "ignore") pdf2image.exceptions.PDFPageCountError: Unable to get page count.

### ISSUE THREAD

**Issue by jjhelfman** [Belval/pdf2image#147]
Attempting to convert pdf to images fails and this is the Traceback:

```
Traceback (most recent call last):
  File "c:\Users\User\Desktop\GIS\PDF to Image\venv\lib\site-packages\pdf2image\pdf2image.py", line 436, in pdfinfo_from_path
    raise ValueError
ValueError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:/Users/User/Desktop/GIS/PDF to Image/pdf2image_script.py", line 9, in <module>        
    images = convert_from_path(["C:\\Users\\User\\Desktop\\GIS\\PDF to Image\\As-Builts 1\\4A 1st Flr B&W.pdf"], 500)
  File "c:\Users\User\Desktop\GIS\PDF to Image\venv\lib\site-packages\pdf2image\pdf2image.py", line 94, in convert_from_path
    page_count = pdfinfo_from_path(pdf_path, userpw, poppler_path=poppler_path)["Pages"]        
  File "c:\Users\User\Desktop\GIS\PDF to Image\venv\lib\site-packages\pdf2image\pdf2image.py", line 446, in pdfinfo_from_path
    "Unable to get page count.\n%s" % err.decode("utf8", "ignore")
pdf2image.exceptions.PDFPageCountError: Unable to get page count.
```

===============================================

This was the python script I ran on VS Code on a Windows 10 machine:

```
from pdf2image import convert_from_path, convert_from_bytes

from pdf2image.exceptions import (
    PDFInfoNotInstalledError,
    PDFPageCountError,
    PDFSyntaxError
)

images = convert_from_path(["C:\\Users\\User\\Desktop\\GIS\\PDF to Image\\As-Builts 1\\4A 1st Flr B&W.pdf"], 500)

for i, image in enumerate(images):
    fname = "image" + str(i) + ".png"
    image.save(fname, "PNG")
```

I am using Python 3.7.4 (64 bit) and have also installed Poppler via Chocolatey after it seemed to lack required dlls when I installed from anaconda. 

Poppler seems to have a lot of issues on Windows, so I began reading up on [another approach](https://towardsdatascience.com/poppler-on-windows-179af0e50150) that seems promising 

Thanks in advance for your help!

**Comment by Belval** [comment]
Thank you for providing a code sample, it helps a lot when troubleshooting on my side.

First this, you shouldn't put the PDF path in square brackets, I am actually surprised that it didn't raise an exception, but the `convert_from_path` function takes a string and not a list.

Then, I think you should try to set the `poppler_path` parameter to see if it solves your issue. For example if my poppler executable directory is in `C:\\Users\Belval\poppler\bin`, I can write:

`convert_from_path("my_file.pdf", dpi=500, fmt="png", poppler_path=r"C:\\Users\Belval\poppler\bin")`

To avoid having to double all your backslashes, you can use the "r" prefix, it will treat the string as "raw" and not escape anything.

**Comment by jjhelfman** [comment]
It works perfectly now and the issue is resolved when I implemented your parameters. As a side note, I discovered that when I went up to 1000 dpi, it gave me the following warning:

`DecompressionBombWarning: Image size (93500000 pixels) exceeds limit of 89478485 pixels, could be decompression bomb DOS attack.
  DecompressionBombWarning,`

Fortunately, 1000 dpi was in fact unnecessary for the particular pdf I was processing. This is an extremely helpful module that I look forward to using the future.
